hasura/metadata/databases/default/tables/public_memberships.yaml

```diff
@@ -0,0 +1,58 @@
+table:
+  name: memberships
+  schema: public
+object_relationships:
+  - name: team
+    using:
+      foreign_key_constraint_on: team_id
+  - name: user
+    using:
+      foreign_key_constraint_on: user_id
+insert_permissions:
+  - role: service
+    permission:
+      check: {}
+      columns:
+        - id
+        - role
+        - team_id
+        - user_id
+select_permissions:
```

Comment:
are we not allowing users to manage memberships on their own? or perhaps we want to tackle permissions for this in a separate PR?

Reply:
What do you mean? Do you want to allow users to change their roles?